### PR TITLE
bind alpha feature network plugin flags correctly

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
When working #63542, I found the flags, like `--cni-conf-dir` and `cni-bin-dir`, were not correctly bound.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/cc kubernetes/sig-node-pr-reviews

**Release note**:

```release-note
None
```

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
> I found the flags, like --cni-conf-dir and cni-bin-dir, were not correctly bound.

Could you elaborate on this?

**Comment by feiskyer** [comment]
By the way, if both flags are not set when starting kubelet (means empty), dockershim will set it when probing CNI plugins [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/network/cni/cni.go).

**Comment by dixudx** [comment]
>>I found the flags, like --cni-conf-dir and cni-bin-dir, were not correctly bound.

>Could you elaborate on this?

https://github.com/kubernetes/kubernetes/blob/2fd9d8654eeb63aed59776e18c012f42b2e6d130/cmd/kubelet/app/options/container_runtime.go#L41-L56

You can see `CNIBinDir` and `CNIConfDir` are not set with default value in `NewContainerRuntimeOptions`, which is causing the bound value to be empty. However we only manually append the default value in the description message.

  > dockershim will set it when probing CNI plugins [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/network/cni/cni.go)

I think we should not set it here. I prefer to set it in `NewContainerRuntimeOptions`, where all the default values stay.

**Comment by dixudx** [comment]
/assign @deads2k 

Gently ping @deads2k for a look.

**Comment by dixudx** [comment]
ping @deads2k for a look. Thanks.

**Review by mtaufen — COMMENTED** [review]
Overall goal of setting defaults in the conventional place sounds good to me.

File: pkg/kubelet/config/flags.go [hunk 1]
```diff
@@ -86,12 +86,12 @@ func (s *ContainerRuntimeOptions) AddFlags(fs *pflag.FlagSet) {
...
 	fs.DurationVar(&s.ImagePullProgressDeadline.Duration, "image-pull-progress-deadline", s.ImagePullProgressDeadline.Duration, fmt.Sprintf("If no pulling progress is made before this deadline, the image pulling will be cancelled. %s", dockerOnlyWarning))
 
 	// Network plugin settings for Docker.
 	fs.StringVar(&s.NetworkPluginName, "network-plugin", s.NetworkPluginName, fmt.Sprintf("<Warning: Alpha feature> The name of the network plugin to be invoked for various events in kubelet/pod lifecycle. %s", dockerOnlyWarning))
-	fs.StringVar(&s.CNIConfDir, "cni-conf-dir", s.CNIConfDir, fmt.Sprintf("<Warning: Alpha feature> The full path of the directory in which to search for CNI config files. Default: /etc/cni/net.d. %s", dockerOnlyWarning))
-	fs.StringVar(&s.CNIBinDir, "cni-bin-dir", s.CNIBinDir, fmt.Sprintf("<Warning: Alpha feature> A comma-separated list of full paths of directories in which to search for CNI plugin binaries. Default: /opt/cni/bin. %s", dockerOnlyWarning))
+	fs.StringVar(&s.CNIConfDir, "cni-conf-dir", s.CNIConfDir, fmt.Sprintf("<Warning: Alpha feature> The full path of the directory in which to search for CNI config files. %s", dockerOnlyWarning))
```
[the remark this answers is not in the document]
Comment (dixudx) [reply pkg/kubelet/config/flags.go]: Cobra will help do this.

[the remark this answers is not in the document]
Comment (mtaufen) [reply pkg/kubelet/config/flags.go]: Can you run `kubelet --help` with this build and double-check, just for my sake?

[the remark this answers is not in the document]
Comment (dixudx) [reply pkg/kubelet/config/flags.go]: Just extract part of `kubelet -h`.

```
      --cni-bin-dir string                                                                                        <Warning: Alpha feature> A comma-separated list of full paths of directories in which to search for CNI plugin binaries. This docker-specific flag only works when container-runtime is set to docker. (default "/opt/cni/bin")
      --cni-conf-dir string                                                                                       <Warning: Alpha feature> The full path of the directory in which to search for CNI config files. This docker-specific flag only works when container-runtime is set to docker. (default "/etc/cni/net.d")
      --config string                                                                                             The Kubelet will load its initial configuration from this file. The path may be absolute or relative; relative paths start at the Kubelet's current working directory. Omit this flag to use the built-in default configuration values. Command-line flags override configuration from this file.
      --container-log-max-files int32                                                                             <Warning: Alpha feature> Set the maximum number of container log files that can be present for a container. The number must be >= 2. (default 5) (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
      --container-log-max-size string                                                                             <Warning: Alpha feature> Set the maximum size (e.g. 10Mi) of container log file before it is rotated. (default "10Mi") (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
      --container-runtime string                                                                                  The container runtime to use. Possible values: 'docker', 'remote', 'rkt (deprecated)'. (default "docker")
```

@mtaufen Cobra will automatically append `(default "xxxValue")` at the end if bound.

[the remark this answers is not in the document]
Comment (mtaufen) [reply pkg/kubelet/config/flags.go]: Thanks, looks good

**Comment by yujuhong** [comment]
/lgtm 

Thanks for the fix and the cleanup!

**Comment by dixudx** [comment]
@yujuhong  @mtaufen Rebased. PTAL. Thanks.

**Review by timothysc — APPROVED** [review]
/approve 

from kubeadm side, I'll let @mtaufen lgtm + approve from that side.

**Pull request merged**